lib/Twig/Extension/Core.php

```diff
@@ -1507,12 +1507,15 @@ function twig_get_attribute(Twig_Environment $env, Twig_Source $source, $object,
                 continue;
             }
 
-            if (!isset($cache[$name])) {
-                $cache[$name] = $method;
-                $cache[$lcName] = $method;
+            if (!isset($classCache[$name])) {
+                $classCache[$name] = $method;
+            }
+
+            if (!isset($classCache[$lcName])) {
```

Comment:
@stof In Twig 1.x the code was as it is in this PR. In #2316 this was changed to the following:

```
if (!isset($classCache[$name])) {
    $classCache[$name] = $method;
    $classCache[$lcName] = $method;
}
```

However, this leads to an issue when you have two methods `bar()` and `getBar()`. When `bar()` has been processed, `$classCache['bar']` has the value `bar`. However, when `getBar()` is processed, `$name` is `Bar`. Since this key is not set in `$classCache`, the `bar` key would have been overwritten too.